pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -420,9 +422,32 @@ public String checkTableConfig(String tableConfigStr) {
       String msg = String.format("Invalid table config json string: %s", tableConfigStr);
       throw new ControllerApplicationException(LOGGER, msg, Response.Status.BAD_REQUEST, e);
     }
+    return validateConfig(tableConfig, _pinotHelixResourceManager.getSchemaForTableConfig(tableConfig));
+  }
+
+  @POST
+  @Path("/tables/validateTableAndSchema")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Validate table config for a table along with specified schema", notes =
+      "Validate given table config and schema. If specified schema is null, attempt to retrieve schema using the "
+          + "table name. This API returns the table config that matches the one you get from 'GET /tables/{tableName}'."
```

Comment:
`This API returns the table config that matches the one you get from 'GET /tables/{tableName}'` - this isn't the case rt? This API returns the same table config that was passed as argument?

Reply:
It does add an extra layer (set TableType - REALTIME/OFFLINE). So in effect, it does return a structure compatible with the GET api.